src/lib/logical_query_plan/join_node.cpp

```diff
@@ -132,10 +132,12 @@ void JoinNode::_update_output() const {
   const auto& right_names = right_child()->output_column_names();
 
   _output_column_names.emplace();
-  _output_column_names->reserve(left_names.size() + right_names.size());
+  bool only_left = _join_mode == JoinMode::Semi || _join_mode == JoinMode::Anti;
+
+  auto output_size = only_left ? left_names.size() : left_names.size() + right_names.size();
```

Comment:
personally I'd prefer `const auto output_column_count = ...` as `output_size` could be anything